### PR TITLE
Fix other ML packages tests

### PR DESCRIPTION
**Pull request by riedgar-ms**
Fix a couple of issues with our tests with other ML packages:

- The latest version of `pandas` objects to how `xgboost` was using it (specifically `FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.`). Temporarily exclude this version of `pandas` with an upper bound
- It appears that `conda-forge` had the latest `scikeras` but not the latest `tensorflow` to match it. Switch to installing both from `pip`

Have filed #1015 as a reminder to watch `xgboost` for an update to be compatible with the latest `pandas`.

### CODE REVIEW

**Comment by riedgar-ms** [comment]
Build: https://dev.azure.com/fairlearn/Fairlearn/_build/results?buildId=1336&view=results

**Review by romanlutz — APPROVED** [review]

**Comment by riedgar-ms** [comment]
Since this is just a build fix, I'll merge on Monday unless @fairlearn/fairlearn-maintainers object.

File: test_othermlpackages/conda-tensorflow.yaml
Comment (adrinjalali) [inline test_othermlpackages/conda-tensorflow.yaml:11]: Or alternatively just `scikeras[tensorflow]`

Comment (riedgar-ms) [reply test_othermlpackages/conda-tensorflow.yaml:11]: Doesn't work for some reason :-(

**Review by adrinjalali — APPROVED** [review]

**Pull request merged**
